sdcm/tester.py

```diff
@@ -1079,6 +1080,16 @@ def clean_resources(self):
 
     def tearDown(self):
         self.clean_resources()
+        self.archive_job_log()
```

Comment:
I think this code won't be reached when we
- have an exception in the `clean_resources`
- exception in setUp

To solve the teardown issue we can solve by making the method as a decorator and inside the decorator catch exception, save it to log, upload and the reraise.
For now, I suggest not to solve the setUp exception issue since we will need to run a watchdog on avocado process and when it finished upload to s3.
@fruch what do you think?

Reply:
@bentsi change a bit of code. adding clean_resource_on_exception decorator to run the zip_and_upload job log. Run several tests by raising the Exception from setUP, tearDown, clean_resources methods.
What do you think?

Reply:
I don't know how your test passed, but `clean_resource_on_exception` is not related to this problem and you will get infinite recursion when there is an exception in `clean_resources` (this decorator runs `clean_resources`) 
I meant to create a new decorator, but after discussing this with @fruch better to do this:

```
def tearDown(self):
    try:
    	self.clean_resources()
    except:
    	self.log.exception("Exception in clean_resources")
    	raise
    finally:
    	self.archive_job_log()
```
Also keep in mind that `archive_job_log` should not fail the test

Reply:
To avoid recursion there was a check in clean_resource_on_exception, that there method are Exception happend is not clean_resource, then run clean_resource, otherwise store the job.log

Reply:
will look the suggested solution